engine/table/src/main/java/io/deephaven/engine/table/impl/select/FilterSelectColumn.java

```diff
@@ -0,0 +1,246 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.chunk.ChunkType;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.WritableObjectChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.exceptions.UncheckedTableException;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.rowset.RowSetFactory;
+import io.deephaven.engine.rowset.TrackingRowSet;
+import io.deephaven.engine.table.*;
+import io.deephaven.engine.table.impl.MatchPair;
+import io.deephaven.engine.table.impl.QueryCompilerRequestProcessor;
+import io.deephaven.engine.table.impl.QueryTable;
+import io.deephaven.engine.table.impl.sources.InMemoryColumnSource;
+import io.deephaven.engine.table.impl.sources.SparseArrayColumnSource;
+import io.deephaven.engine.table.impl.sources.ViewColumnSource;
+import io.deephaven.qst.column.header.ColumnHeader;
+import io.deephaven.util.type.TypeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The FilterSelectColumn wraps a {@link io.deephaven.api.filter.Filter} and producing a column of true or false Boolean
+ * values described by the filter.
+ *
+ * <p>
+ * This column is appropriate as an argument to a {@link Table#view(Collection)} or {@link Table#updateView(String...)},
+ * lazily evaluating the equivalent of a {@link Table#wouldMatch(String...)} operation. Although select and update can
+ * also use Filters, the wouldMatch provides a more efficient path for realized results as it stores and updates only a
```

Comment:
Or rather, wouldMatch would just delegate to the this new SelectColumn.

Reply:
The select/update code is very different in terms of how it iterates the rows; does dependency tracking, etc. so I don't think trying to unify the implementations makes sense.

Reply:
Should we consider running a wouldMatch when the user uses update/select w/ Filter? The easier cases would be when they are used standalone:

```
t.update(Filter.of(...)) ~= t.wouldMatch(...)
t.select(Filter.of(...)) ~= t.wouldMatch(...).view(output)
```

The slightly more complicated would be when mixed w/ other SelectColumns; assume the column don't depend on one another:

```
t.update(A, Filter.of(...), C) ~= t.wouldMatch(...).update(A, C).moveColumns(...)
t.select(A, Filter.of(...), C) ~= t.wouldMatch(...).update(A, C).view(A, ..., C)
```

I'm fine with "no" for now (or "no" forever b/c there is are some semantic differences), just trying to make sure I understand.

Reply:
We have some precedent with that in where, there are some magic "reindexing" filters that we examine and break the where up into a variety of operations.  If we were to before actually doing the select, shunt things off to wouldMatch if present it would work.  There are two things that I see potentially causing problems:
1. wouldMatch is not parallelized, and we could end up reducing parallelization of other columns.
2. The resulting column source is "in-memory" but not one of our in-memory column sources.  I'm not sure if we should think that is violating the contract or not.

So my general take is that it is probably enough complexity that we should defer it until it matters one day, and only then take it on.